st2auth/st2auth/handlers.py

```diff
@@ -130,6 +130,25 @@ def handle_auth(
         remote_addr = headers.get("x-forwarded-for", remote_addr)
         extra = {"remote_addr": remote_addr}
 
+        # Needed to support st2client which does not connect via st2web
+        if authorization and not remote_user:
+            try:
+                auth_value = base64.b64decode(authorization[1])
+            except Exception:
+                LOG.audit("Invalid authorization header", extra=extra)
+                abort_request()
+                return
+
+            split = auth_value.split(b":", 1)
+            if len(split) != 2:
+                LOG.audit("Invalid authorization header", extra=extra)
+                abort_request()
+                return
+
+            remote_user = split[0]
+            if six.PY3 and isinstance(remote_user, six.binary_type):
```

Comment:
Do we need to have checks on whether its python3 any more?

Reply:
Probably not but i was just keeping consistent with what is used in other handlers, for example: https://github.com/StackStorm/st2/blob/32a243a2ed54c5233e10f08b69f0cb21e2f10cb7/st2auth/st2auth/handlers.py#L212-L216

`decode("utf-8")` is needed even without the conditional in my testing.